File: Windows/SibeliusReset/Themes/DesignTokens.cs
using System;
using System.Windows.Media;

namespace SibeliusReset.Themes
{
    /// <summary>
    /// Application identity constants.
    /// </summary>
    public static class AppIdentity
    {
        public const string DisplayName = "Sibelius重置";
    }

    /// <summary>
    /// Application color palette ported from the macOS SwiftUI design.
    /// All colors are defined as <see cref="Color"/> properties with corresponding
    /// <see cref="SolidColorBrush"/> helpers via the <see cref="Brush"/> method.
    /// </summary>
    public static class AppColor
    {
        // ── Backgrounds ──────────────────────────────────────────────

        public static Color DeepSpaceBlack  => ColorFromHex("#050A18");
        public static Color MidnightNavy    => ColorFromHex("#08111F");
        public static Color PanelBlueBlack  => ColorFromHex("#101A33");

        // ── Accent / Brand ───────────────────────────────────────────

        public static Color ElectricCyan    => ColorFromHex("#38E8FF");
        public static Color LaserBlue       => ColorFromHex("#297BFF");
        public static Color VioletPurple    => ColorFromHex("#7B61FF");
        public static Color NeonMagenta     => ColorFromHex("#FF4DFF");
        public static Color SoftAqua        => ColorFromHex("#7DF9FF");

        // ── Text ─────────────────────────────────────────────────────

        public static Color PrimaryText     => ColorFromHex("#F4F7FF");
        public static Color SecondaryText   => ColorFromHex("#A9B6D3");
        public static Color MutedText       => ColorFromHex("#6F7EA3");

        // ── Semantic ─────────────────────────────────────────────────

        public static Color WarningAmber    => ColorFromHex("#FFCC66");
        public static Color DangerPinkRed   => ColorFromHex("#FF5C8A");

        // ── Helpers ──────────────────────────────────────────────────

        /// <summary>
        /// Creates a frozen <see cref="SolidColorBrush"/> from the given <see cref="Color"/>.
        /// Freezing the brush improves rendering performance in WPF.
        /// </summary>
        public static SolidColorBrush Brush(Color color)
        {
            var brush = new SolidColorBrush(color);
            brush.Freeze();
            return brush;
        }

        /// <summary>
        /// Parses a hex color string (e.g. "#FF5C8A" or "#80FF5C8A") into a <see cref="Color"/>.
        /// Supports #RGB, #RRGGBB, and #AARRGGBB formats.
        /// </summary>
        public static Color ColorFromHex(string hex)
        {
            if (string.IsNullOrWhiteSpace(hex))
                throw new ArgumentException("Hex color string must not be null or empty.", nameof(hex));

            hex = hex.TrimStart('#');

            byte a = 255, r, g, b;

            switch (hex.Length)
            {
                case 3: // #RGB
                    r = Convert.ToByte(new string(hex[0], 2), 16);
                    g = Convert.ToByte(new string(hex[1], 2), 16);
                    b = Convert.ToByte(new string(hex[2], 2), 16);
                    break;
                case 6: // #RRGGBB
                    r = Convert.ToByte(hex.Substring(0, 2), 16);
                    g = Convert.ToByte(hex.Substring(2, 2), 16);
                    b = Convert.ToByte(hex.Substring(4, 2), 16);
                    break;
                case 8: // #AARRGGBB
                    a = Convert.ToByte(hex.Substring(0, 2), 16);
                    r = Convert.ToByte(hex.Substring(2, 2), 16);
                    g = Convert.ToByte(hex.Substring(4, 2), 16);
                    b = Convert.ToByte(hex.Substring(6, 2), 16);
                    break;
                default:
                    throw new FormatException($"Invalid hex color format: #{hex}");
            }

            return Color.FromArgb(a, r, g, b);
        }
    }

    /// <summary>
    /// Application sizing constants for window, controls, and layout elements.
    /// Values are in device-independent pixels (DIPs).
    /// </summary>
    public static class AppSize
    {
        // ── Window ───────────────────────────────────────────────────

        public const double WindowWidth  = 420;
        public const double WindowHeight = 520;
        public const double WindowRadius = 28;

        // ── Info Panel ───────────────────────────────────────────────

        public const double InfoPanelRadius = 18;
        public const double InfoPanelWidth  = 340;
        public const double InfoPanelHeight = 72;

        // ── Buttons ──────────────────────────────────────────────────

        public const double ButtonRadius       = 32;
        public const double PrimaryButtonWidth  = 300;
        public const double PrimaryButtonHeight = 56;

        // ── Countdown Ring ───────────────────────────────────────────

        public const double CountdownOuterSize = 248;
        public const double CountdownRingSize  = 210;
        public const double CountdownInnerSize = 162;
    }
}

File: Sources/DesignTokens.swift
import SwiftUI

enum AppIdentity {
    static let displayName = "Sibelius重置"
}

enum AppColor {
    static let deepSpaceBlack = Color(nsColor: NSColor(hex: "#050A18") ?? .black)
    static let midnightNavy = Color(nsColor: NSColor(hex: "#08111F") ?? .black)
    static let panelBlueBlack = Color(nsColor: NSColor(hex: "#101A33") ?? .black)

    static let electricCyan = Color(nsColor: NSColor(hex: "#38E8FF") ?? .cyan)
    static let laserBlue = Color(nsColor: NSColor(hex: "#297BFF") ?? .blue)
    static let violetPurple = Color(nsColor: NSColor(hex: "#7B61FF") ?? .purple)
    static let neonMagenta = Color(nsColor: NSColor(hex: "#FF4DFF") ?? .magenta)
    static let softAqua = Color(nsColor: NSColor(hex: "#7DF9FF") ?? .cyan)

    static let primaryText = Color(nsColor: NSColor(hex: "#F4F7FF") ?? .white)
    static let secondaryText = Color(nsColor: NSColor(hex: "#A9B6D3") ?? .gray)
    static let mutedText = Color(nsColor: NSColor(hex: "#6F7EA3") ?? .gray)

    static let warningAmber = Color(nsColor: NSColor(hex: "#FFCC66") ?? .yellow)
    static let dangerPinkRed = Color(nsColor: NSColor(hex: "#FF5C8A") ?? .red)
}

enum AppMotion {
    static let entrance = Animation.spring(response: 0.72, dampingFraction: 0.86, blendDuration: 0.08)
    static let softSpring = Animation.spring(response: 0.46, dampingFraction: 0.82, blendDuration: 0.08)
    static let statusChange = Animation.easeInOut(duration: 0.34)
}

enum AppSize {
    static let windowWidth: CGFloat = 420
    static let windowHeight: CGFloat = 520

    static let windowRadius: CGFloat = 28
    static let infoPanelRadius: CGFloat = 18
    static let buttonRadius: CGFloat = 32

    static let countdownOuterSize: CGFloat = 248
    static let countdownRingSize: CGFloat = 210
    static let countdownInnerSize: CGFloat = 162

    static let primaryButtonWidth: CGFloat = 300
    static let primaryButtonHeight: CGFloat = 56

    static let infoPanelWidth: CGFloat = 340
    static let infoPanelHeight: CGFloat = 72
}

enum AppFont {
    static let title = Font.system(size: 22, weight: .bold, design: .default)
    static let status = Font.system(size: 14, weight: .medium, design: .default)

    static let circleLabel = Font.system(size: 16, weight: .semibold, design: .default)
    static let mainNumber = Font.system(size: 74, weight: .heavy, design: .rounded)
    static let dayUnit = Font.system(size: 28, weight: .semibold, design: .default)
    static let circleSubtext = Font.system(size: 15, weight: .medium, design: .default)

    static let button = Font.system(size: 21, weight: .bold, design: .default)
    static let info = Font.system(size: 13, weight: .medium, design: .default)
    static let footer = Font.system(size: 13, weight: .medium, design: .default)
}

extension NSColor {
    convenience init?(hex: String) {
        let r, g, b: CGFloat
        var hexColor = hex
        if hexColor.hasPrefix("#") {
            hexColor.remove(at: hexColor.startIndex)
        }
        if hexColor.count == 6 {
            let scanner = Scanner(string: hexColor)
            var hexNumber: UInt64 = 0
            if scanner.scanHexInt64(&hexNumber) {
                r = CGFloat((hexNumber & 0xff0000) >> 16) / 255
                g = CGFloat((hexNumber & 0x00ff00) >> 8) / 255
                b = CGFloat(hexNumber & 0x0000ff) / 255
                self.init(red: r, green: g, blue: b, alpha: 1.0)
                return
            }
        }
        return nil
    }
}
